webapp/packages/core-root/src/SessionResource.ts

```diff
@@ -73,6 +58,25 @@ export class SessionResource extends CachedDataResource<SessionState | null> {
     this.defaultLocale = defaultLocale;
   }
 
+  private handleSessionStateEvent(event: ISessionStateEvent) {
+    if (!this.data) {
+      return;
+    }
+
+    const sessionState: SessionState = {
+      ...this.data,
+      valid: event?.isValid ?? this.data?.valid,
+      remainingTime: event.remainingTime,
+      actionParameters: event.actionParameters,
+      cacheExpired: event?.isCacheExpired ?? this.data.cacheExpired,
+      lastAccessTime: String(event.lastAccessTime),
+      locale: event.locale,
+    };
+
+    this.setData(sessionState);
```

Comment:
seems like we need to prohibit using setData outside of `loader` or `performUpdate` because it will change data but event's won't be fired

Reply:
<img width="774" alt="image" src="https://github.com/dbeaver/cloudbeaver/assets/26230285/cae4beb3-fae3-4a04-a746-c2d72a7d61e7">